### PR TITLE
Add helm chart for deploying k8s dev environment

### PR DESCRIPTION
**Pull request by santoshamohan**
- Add `helm` chart template for deploying `k8s` `dev` environment

### CODE REVIEW

File: stable/datacube-dev-env/values.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  registry: docker.io
+  repository: opendatacube/k8s-dev
+  tag: "latest"
+  pullPolicy: Always
+
+datacube:
+  configPath: /opt/odc/datacube.conf
+
+database:
+  create: true
```
Comment (harshurampur) [inline stable/datacube-dev-env/values.yaml]: Better to keep the defaults  set to False

Comment (santoshamohan) [reply stable/datacube-dev-env/values.yaml]: Updated

File: stable/datacube-dev-env/templates/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    - host: "{{ . }}.{{ $domain }}"
+      http:
+        paths:
+          - path: {{ $ingressPath }}
+            backend:
+              serviceName: {{ $fullName }}
+              servicePort: {{ $servicePort }}
+  {{- end }}
+  {{- end }}
+{{- end }}
```
Comment (tom-butler) [inline stable/datacube-dev-env/templates/ingress.yaml]: Ingress is not required for a dev container

File: stable/datacube-dev-env/templates/install-postgis.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            valueFrom:
+              secretKeyRef:
+                name: {{ .Values.database.adminSecret }}
+                key: postgres-username
+          - name: ADMIN_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: {{ .Values.database.adminSecret }}
+                key: postgres-password
+      restartPolicy: Never
```
Comment (tom-butler) [inline stable/datacube-dev-env/templates/install-postgis.yaml]: Potsgis extension shouldn't be required either, (they can just run the command if needed)

File: stable/datacube-dev-env/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+          imagePullPolicy: Always
+          env:
+          - name: DB_HOSTNAME
+            value: {{ .Values.database.host | quote }}
+          - name: DB_PORT
+            value: {{ .Values.database.port | quote }}
+          - name: DB_DATABASE
+            value: {{ .Values.database.database | quote }}
+          - name: REF_DATABASE
+            value: {{ .Values.database.ref_database | quote }}
```
Comment (tom-butler) [inline stable/datacube-dev-env/templates/deployment.yaml]: What is ref database for?

File: stable/datacube-dev-env/values.yaml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  periodSeconds: 10
+
+## Define readiness probe
+## For a readiness probe, giving up means not routing traffic to the pod, but the pod is not restarted.
+readinessProbe:
+  exec:
+    command:
+      - "ls"
+  initialDelaySeconds: 5
+  periodSeconds: 10
```
Comment (tom-butler) [inline stable/datacube-dev-env/values.yaml]: These probes have been removed, their values should be too

File: stable/datacube-dev-env/values.yaml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  dockerArgs:
+    - "/bin/bash"
+    - "-c"
+    - "echo 'CREATE EXTENSION IF NOT EXISTS postgis; \n grant agdc_admin to $ADMIN_USERNAME;' |
+       PGPASSWORD=$ADMIN_PASSWORD psql -h $DB_HOSTNAME -p 5432 -U $ADMIN_USERNAME -d $DB_DATABASE"
+  image:
+    registry: docker.io
+    repository: postgres
+    tag: latest
+    pullPolicy: Always
```
Comment (tom-butler) [inline stable/datacube-dev-env/values.yaml]: This should be removed with the postgis setup

File: stable/datacube-dev-env/values.yaml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+containerPort: 8080 # Internal port on which pod is running
+
+ingress:
+  ## Enable Ingress.
+  ##
+  enabled: false
+  host:
```
Comment (tom-butler) [inline stable/datacube-dev-env/values.yaml]: this should be removed with the ingress

File: stable/datacube-dev-env/templates/ingress.yaml (hunk 2 above)
Comment (santoshamohan) [reply stable/datacube-dev-env/templates/ingress.yaml]: Ingress file removed

File: stable/datacube-dev-env/templates/install-postgis.yaml (hunk 3 above)
Comment (santoshamohan) [reply stable/datacube-dev-env/templates/install-postgis.yaml]: Postgis extension template file removed

File: stable/datacube-dev-env/templates/deployment.yaml (hunk 4 above)
Comment (santoshamohan) [reply stable/datacube-dev-env/templates/deployment.yaml]: Removed redundant ref database

File: stable/datacube-dev-env/values.yaml (hunk 5 above)
Comment (santoshamohan) [reply stable/datacube-dev-env/values.yaml]: Removed liveness probe and readiness probe from the values.yaml file

File: stable/datacube-dev-env/values.yaml (hunk 6 above)
Comment (santoshamohan) [reply stable/datacube-dev-env/values.yaml]: Removed postgis setup from values.yaml file

File: stable/datacube-dev-env/values.yaml (hunk 7 above)
Comment (santoshamohan) [reply stable/datacube-dev-env/values.yaml]: Removed ingress values

**Pull request merged**